vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go

```diff
@@ -24,6 +24,7 @@ import (
 	"strings"
 
 	"k8s.io/kube-openapi/pkg/util/proto"
+	"k8s.io/kube-openapi/pkg/validation/spec"
```

Comment:
Can you make the PR against `kube-openapi`, I would *maybe* do it in a separate package just so that it doesn't pull the proto dependency but I would definitely keep the existing code in-place since you're writing a complete replacement.

Reply:
Yea, I have a pending branch just no PR yet while I work on it.

Regarding code sharing, there was a lot of churn, I will make a pass to clean up the diff at the end, but I think there will be code sharing for the logic which interprets the x-kubernetes* extensions, since I think to maintain duplicate copies might be error-prone

Reply:
But we probably don't want to maintain duplicate copies, we just want to eventually be able to `git rm old-schemaconv` and call it done. Tying both together makes that task a lot more tedious.